src/components/GroceryItem/GroceryItem.js

```diff
@@ -47,16 +48,49 @@ const GroceryItem = ({ item }) => {
       });
   };
 
-  const resetDate = () => {
+  // Should we delete the option to uncheck? Otherwise, we can't really reset the previousPurchaseDate in Firestore.
+  const handleUncheckBox = () => {
     setChecked(false);
   };
 
+  const getAriaLabel = () => {
+    if (
+      currentDate - item.purchaseDate >=
+        2 * (item.nextPurchaseDate - item.purchaseDate) ||
+      item.numberOfPurchases === 0
```

Comment:
**Non-blocking:** I see that you use this condition here and inside `getBackground` (lines 74-76). What do you think about abstracting this into a separate function? That would make it easier in case you have to make a change to that logic later.